### ISSUE TITLE
Allow clipping in facet labels

### ISSUE THREAD

**Issue by malcolmbarrett** [tidyverse/ggplot2#3684]
By default, ggplot2 reasonably clips long facet labels. While this can be turned off by manipulating the underlying grobs, it would be nice for some circumstances if you could set `clip = "off"` in a related ggplot2 function, as you can in the `coord_*()` functions.

``` r
library(ggplot2)
iris$Species <- factor(
  iris$Species, 
  labels = c(
    "Here is a pretty long label that might not make it, if we're being totally honest", 
    "here's a shorter one", 
    "shorter yet"
  )
)

ggplot(iris, aes(Sepal.Length, Sepal.Width)) +
  geom_point() + 
  facet_wrap(vars(Species))
```

![](https://i.imgur.com/Pbl92YN.png)

<sup>Created on 2019-12-20 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by thomasp85** [comment]
It would be quite easy to make this a theme setting (I have just fiddled with that code in another matter). What say the @clauswilke, @Hadley, do you have any objections to this?

**Comment by clauswilke** [comment]
I think it's a good idea.

**Comment by thomasp85** [comment]
This change intersects with #3683 so I'll wait for that to get merged

**Comment by teunbrand** [comment]
This is fixed by #4223, though the long label is obscured on the right by the strip background of the middle panel.

``` r
library(ggplot2)
iris$Species <- factor(
  iris$Species, 
  labels = c(
    "Here is a pretty long label that might not make it, if we're being totally honest", 
    "here's a shorter one", 
    "shorter yet"
  )
)

ggplot(iris, aes(Sepal.Length, Sepal.Width)) +
  geom_point() + 
  facet_wrap(vars(Species)) +
  theme(strip.clip = "off")
```

![](https://i.imgur.com/wN8CeDo.png)

<sup>Created on 2022-05-20 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
